tests/test_web_server.py

```diff
@@ -31,13 +37,29 @@ async def test_unsupported_upgrade(aiohttp_raw_server, aiohttp_client) -> None:
     # don't fail if a client probes for an unsupported protocol upgrade
     # https://github.com/aio-libs/aiohttp/issues/6446#issuecomment-999032039
     async def handler(request: web.Request):
-        return web.Response(body=await request.read())
+        try:
+            import pprint
+
+            pprint.pprint(["handler called"])
+            async with async_timeout.timeout(10):
+                result = await request.read()
+            pprint.pprint(["handler read", result])
+            return web.Response(body=result)
+        except Exception as e:
+            import pprint
+
+            pprint.pprint(["handler except", e])
+            raise
 
     upgrade_headers = {"Connection": "Upgrade", "Upgrade": "unsupported_proto"}
```

Comment:
I'm not sure what is parsing the `Upgrade` header, but the fact that it's not a valid token at all should also be considered ([IANA upgrade token registry](https://www.iana.org/assignments/http-upgrade-tokens/http-upgrade-tokens.xhtml).  In fact, it even contains an invalid character, "_", for URL schemes ([RFC 1738](https://datatracker.ietf.org/doc/html/rfc1738#section-2.1)).

Reply:
I would assume that would be handled in the parsers (i.e. llhttp or http_parser.py, depending on whether extensions are enabled).

Reply:
Ah okay.  My point is, especially it it happens externally, the test should know and account for the behavior.

I would hope, no matter which parser is used, that "unsupported_proto" would simply be thrown away and equivalent to "".  Then the test should also test a valid, but unsupported, token.